Lib/socketserver.py

```diff
@@ -651,24 +652,32 @@ def process_request_thread(self, request, client_address):
         except Exception:
             self.handle_error(request, client_address)
         finally:
-            self.shutdown_request(request)
+            try:
+                self.shutdown_request(request)
+            finally:
+                thread = threading.current_thread()
```

Comment:
Can you please put this code into a private sub-method, something like: self._remove_thread()?

Reply:
In the latest commits, this behavior is moved to `_threads.remove_current()`.